File: pdui-express/lib/core/PDUIGetFormData.ts
import { PDUIFn } from "../models/PDUIFn";
import type { PBFnExpression } from "../proto-out/pdui-proto-out";

export class PDUIGetFormData extends PDUIFn {
    formKey: string;

    constructor({ formKey }: { formKey: string }) {
        super({
            identity: "getFormData",
            params: {},
            body: {},
        });

        this.formKey = formKey;
    }

    override toPB() {
        return { formKey: this.formKey };
    }

    override toFnExpression(): PBFnExpression {
        return {
            identity: "getFormData",
            payload: {
                oneofKind: "getFormData",
                getFormData: { formKey: this.formKey },
            },
        };
    }
}

File: pdui_flutter/lib/core/pdui_get_form_data.dart
import 'package:flutter/material.dart';
import 'package:pdui_flutter/core/pdui_context.dart';

import '../models/pdui_fn.dart';
import '../proto_out/pdui-proto-out.pb.dart';
import '../widgets/pdui_form.dart';

class PduiGetFormData extends PduiFn {
  PBGetFormData pbGetFormData;

  PduiGetFormData(this.pbGetFormData) : super("getFormData");

  @override
  resolve() {
    debugPrint(
      PduiFormController.formsInputValues[pbGetFormData.formKey].toString(),
    );

    //TODO: Remove this
    showDialog(
      context: PduiContext.navigatorKey.currentContext!,
      builder: (BuildContext context) {
        return AlertDialog(
          title: Text("Form data"),
          content: Text(
            PduiFormController.formsInputValues[pbGetFormData.formKey]
                .toString(),
          ),
          actions: <Widget>[
            ElevatedButton(
              child: const Text("Ok"),
              onPressed: () =>
                  Navigator.of(PduiContext.navigatorKey.currentContext!).pop(),
            ),
          ],
        );
      },
    );

    return PduiFormController.formsInputValues[pbGetFormData.formKey];
  }
}
